File: src/main/java/com/openvalue/boardgameratings/service/boardgame/BoardGameEntity.java
package com.openvalue.boardgameratings.service.boardgame;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Getter
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PACKAGE)
public class BoardGameEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", nullable = false)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(name = "category", nullable = false)
    private Category category;

    @Column(name = "minNrOfPlayers", nullable = false)
    private Integer minimalNumberOfPlayers;
    
    @Column(name = "maxNrOfPlayers", nullable = false)
    private Integer maximalNumberOfPlayers;
    
    @Column(name = "minimalAge", nullable = false)
    private Integer minimalAge;
    
    @Column(name = "maximalAge", nullable = false)
    private Integer maximalAge;

}

File: src/main/java/com/openvalue/boardgameratings/service/boardgame/BoardGameEntity.kt
package com.openvalue.boardgameratings.service.boardgame

import lombok.AccessLevel
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import javax.persistence.*


@Entity
data class BoardGameEntity (

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long?,

    @Column(name = "name", nullable = false)
    val name: String,

    @Enumerated(EnumType.STRING)
    @Column(name = "category", nullable = false)
    val category: Category,

    @Column(name = "minNrOfPlayers", nullable = false)
    val minimalNumberOfPlayers: Int,

    @Column(name = "maxNrOfPlayers", nullable = false)
    val maximalNumberOfPlayers: Int,

    @Column(name = "minimalAge", nullable = false)
    val minimalAge: Int,

    @Column(name = "maximalAge", nullable = false)
    val maximalAge: Int

)